src/threads/thread.h

```diff
@@ -80,22 +82,50 @@ typedef int tid_t;
    only because they are mutually exclusive: only a thread in the
    ready state is on the run queue, whereas only a thread in the
    blocked state is on a semaphore wait list. */
+struct tcb
+{
+  tid_t tid;
+  bool exit;
+  bool wait;
+  bool goa;
+  int exit_code;
+  char * argv;
+  char * prog;
+  struct thread * me;
+  struct thread * parent;
+  struct list_elem elem;
+  struct semaphore sema;
+  struct semaphore wait_sema;
+};
+struct list all_list;
+struct filedescriptor
+{
+  int fd_num;
+  struct file * f;
+  struct thread * master;
+  struct list_elem elem;
+};
```

Comment:
Our fileDescriptor Struct has an ID number , a file struct to orchestrate methods such as write, close, open, writeat(), etc... a thread parent called master, and a list to keep track of